### DISCUSSION TITLE
Custom action: whitespace in input parameters is getting trimmed

### DISCUSSION THREAD

**Discussion by frabert** [community/community#27047] [Actions]
<p>I’ve developed an action that runs regex substitution on input strings, and today I’ve received a bug report that it is not possible to replace strings with whitespace: <a href="https://github.com/frabert/replace-string-action/issues/6" class="inline-onebox" rel="noopener nofollow ugc">not able to replace char with whitespace · Issue #6 · frabert/replace-string-action · GitHub</a></p>
<p>From my brief testing it looks like <code>core.getInput()</code> behaves as if it automatically performs a <code>trim()</code> on the data. Is there any way to turn this behavior off?</p>

**Comment by Simran-B** [comment]
<p>I wasn’t sure if this could be a problem with YAML itself, but I’m fairly sure after some testing that the <code>@actions/core</code> package or something else provided by GitHub strips whitespace from action inputs. <code>replace-with: "\u0020"</code> (with double quotes) doesn’t work, even though that would be the way to escape something in YAML. The only solution that I can think of would be to add another input to let the action know that the input is JSON-encoded, pass the space in like <code>replace-with: '" "'</code> and let the action decode this input with <code>JSON.parse()</code>.</p>

**Comment by Simran-B** [comment, marked as answer]
<p>There’s actually a better way - <code>@actions/core</code> supports an option to turn trimming of whitespace off since v1.3.0: <a href="https://github.com/actions/toolkit/pull/802" class="inline-onebox" rel="noopener nofollow ugc">Core: Add trimWhiteSpace to getInput by luketomlinson · Pull Request #802 · actions/toolkit · GitHub</a></p>
<pre><code class="lang-auto">core.getInput('&lt;name&gt;', { trimWhitespace: false })
</code></pre>
